### PR TITLE
v0.8.0 integration Deposit Contract Updates

### PR DESCRIPTION
**Pull request by dangerousfood**
Resolves Issue #772 
`DepositContract` class was updated to reflect the changes in `v0.8.0`

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java [hunk 1]
```diff
@@ -114,6 +119,74 @@ public static BeaconStateWithCache get_genesis_beacon_state(
     return state;
   }
 
+  public static BeaconState initialize_beacon_state_from_eth1(
+      Bytes32 eth1_block_hash, UnsignedLong eth1_timestamp, List<Deposit> deposits) {
+    UnsignedLong genesis_time =
+        eth1_timestamp.minus(
```
Comment (cemozerr) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java:125]: I'm not sure if it matters, but here minus should be before addition

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java [hunk 2]
```diff
@@ -114,6 +119,74 @@ public static BeaconStateWithCache get_genesis_beacon_state(
...
+        .forEach(
+            index -> {
+              state.getActive_index_roots().set(index, active_index_root);
+              state.getCompact_committees_roots().set(index, committee_root);
+            });
+    return state;
+  }
+
+  public static boolean is_valid_genesis_state(BeaconState state) {
+    return !(state.getGenesis_time().compareTo(MIN_GENESIS_TIME) >= 0)
```
Comment (cemozerr) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java]: This returns true when state genesis time is less than min genesis time. It should return false.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java [hunk 3]
```diff
@@ -71,36 +77,87 @@ public StateProcessor(ServiceConfig config, ChainStorageClient chainStorageClien
     this.chainStorageClient = chainStorageClient;
   }
 
-  @Subscribe
-  public void onEth2GenesisEvent(Eth2GenesisEvent event) {
-    STDOUT.log(
-        Level.INFO,
-        "******* Eth2Genesis Event detected ******* : "
-            + ((Eth2Genesis) event).getDeposit_root().toString());
+  public void onEth2Genesis(BeaconState initial_state) {
```
Comment (cemozerr) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java:80]: It seems like you're reverting to the old version of onEth2Genesis event here. We should keep the new version if we can.

**Review by cemozerr — CHANGES_REQUESTED** [review]
Made a few requests in comments. If you can change them that would be great.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java (hunk 3 above)
Comment (dangerousfood) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java:80]: Not sure what this means. There is no onEth2Genesis event anymore. There is no Eth2Genesis event in the new spec. We detect it ourselves.

Comment (cemozerr) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java:80]: I mean you're reverting to the old way Eth2GenesisEvent function was handled. The changes in the spec forced us to change this. So I'm just saying keep the changes we made.

Comment (dangerousfood) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java:80]: I think you should look more closely at the method and where it is used.

**Review by cemozerr — APPROVED** [review]
LGTM.

**Pull request merged**